apps/hyd/BUILD

```diff
@@ -0,0 +1,121 @@
+load("@aspect_rules_js//js:defs.bzl", "js_library")
+load("@aspect_rules_ts//ts:defs.bzl", "ts_config")
+load("@bazel_skylib//rules:build_test.bzl", "build_test")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("@npm//apps/hyd:next/package_json.bzl", next_bin = "bin")
+
+# TODO: No clue why `apps/hyd:tailwindcss` doesn't exist
+load("@npm//blog:tailwindcss/package_json.bzl", tailwind_bin = "bin")
+load("@rules_oci//oci:defs.bzl", "oci_image_index", "oci_push")
+load("//bzl/macros:js_image.bzl", "js_image")
+load("//bzl/macros:next.bzl", "next")
+
+npm_link_all_packages(name = "node_modules")
+
+_SRCS = [
+    "//apps/hyd/app",
+    "//apps/hyd/lib",
+    "//apps/hyd/styles",
+]
+
+next_bin.next_binary(
+    name = "next_js_binary",
+    visibility = ["//visibility:public"],
+)
+
+ts_config(
+    name = "tsconfig",
+    src = "tsconfig.json",
+    visibility = ["//visibility:public"],
+)
+
+platforms = [
+    ("arm64", "//bzl/platforms:linux_arm64"),
+    ("amd64", "//bzl/platforms:linux_amd64"),
+]
+
+[
+    next(
+        name = "next_{}".format(platform_name),
+        srcs = _SRCS + [
+            ":tailwind",
+        ] + glob(
+            [
+                ".env*",
+            ],
+        ),
+        data = [
+            "next.config.ts",
+            "tsconfig.json",
+            ":node_modules/@tailwindcss/postcss",
+            ":node_modules/autoprefixer",
+            ":node_modules/next",
+            ":node_modules/pg",
+            ":node_modules/react",
+            ":node_modules/react-dom",
+            ":node_modules/tailwindcss",
+            ":node_modules/typescript",
+            "//apps/hyd/public",
+        ],
+        next_bin = "./node_modules/next/dist/bin/next",
+        next_js_binary = ":next_js_binary",
+        node_toolchain = "@nodejs_linux_{}//:toolchain".format(platform_name),
```

Comment:
todo: file ticket when I'm in a better mood to write it nicely

https://bazelbuild.slack.com/archives/CEZUUKQ6P/p1704633856101699?thread_ts=1704565398.412829&cid=CEZUUKQ6P